azure_iot_edge/README.md

```diff
@@ -0,0 +1,146 @@
+        - `ExperimentalFeatures__Enabled`: `true`
+        - `ExperimentalFeatures__EnableMetrics`: `true`
+
+1. Configure the **Edge Hub** runtime module as follows:
+    - Image version must be `1.0.10` or above.
+    - Under "Environment Variables", experimental metrics must be enabled by adding these environment variables (note the double underscores):
+        - `ExperimentalFeatures__Enabled`: `true`
+        - `ExperimentalFeatures__EnableMetrics`: `true`
+
+1. Install and configure the Datadog Agent as a **custom module**:
+    - Set the module name. For example: `datadog-agent`.
+    - Set the Agent image URI. For example: `datadog/agent:7.24.0`.
+    - Under "Environment Variables", configure your `DD_API_KEY`. You may also set extra Agent configuration here (see [Agent Environment Variables][4]).
+    - Under "Container Create Options", enter the following configuration based on your device OS. Edit the `com.datadoghq.ad.instances` option as appropriate. See the [sample azure_iot_edge.d/conf.yaml][5] for all available configuration options. See the documentation on [Docker Integrations Autodiscovery][6] for more information on labels-based integration configuration. **Note**: `NetworkId` must correspond to the network name set in the device `config.yaml` file.
+        - Linux:
+            ```json
+            {
+                "HostConfig": {
+                    "NetworkMode": "default",
+                    "Env": ["NetworkId=azure-iot-edge"],
+                    "Binds": [
+                        "/var/run/docker.sock:/var/run/docker.sock",
+                        "/var/run/iotedge/mgmt.sock:/var/run/iotedge/mgmt.sock"
+                    ]
+                },
+                "Labels": {
+                    "com.datadoghq.ad.check_names": "[\"azure_iot_edge\"]",
+                    "com.datadoghq.ad.init_configs": "[{}]",
+                    "com.datadoghq.ad.instances": "[{\"edge_hub_prometheus_url\": \"http://edgeHub:9600/metrics\", \"edge_agent_prometheus_url\": \"http://edgeAgent:9600/metrics\", \"security_manager_management_api_url\": \"unix:///var/run/iotedge/mgmt.sock\"}]"
+                }
+            }
+            ```
+        - Windows:
+            ```json
+            {
+                "HostConfig": {
+                    "NetworkMode": "default",
+                    "Env": ["NetworkId=azure-iot-edge"],
+                    "Binds": ["/var/run/docker.sock:/var/run/docker.sock"]
+                },
+                "Labels": {
+                    "com.datadoghq.ad.check_names": "[\"azure_iot_edge\"]",
+                    "com.datadoghq.ad.init_configs": "[{}]",
+                    "com.datadoghq.ad.instances": "[{\"edge_hub_prometheus_url\": \"http://edgeHub:9600/metrics\", \"edge_agent_prometheus_url\": \"http://edgeAgent:9600/metrics\", \"security_manager_management_api_url\": \"http://host.docker.internal:15580/\"]}]"
```

Comment:
🤔 Need to think about it. Interestingly this worked on my Linux VM too, so…? But you're right, this is not documented anywhere for Docker on Linux (at least not that I could find).

I think there's a small discussion to be had related to this "check the security manager health by pinging on its management API running on the host" idea, though… And whether we need a service check for the security manager at all, at least in an initial release, since that's proving to cause so many configuration / compatibility problems. 🤔

Reply:
Dropped the security manager service check, as per offline discussion!